db/crud.go

```diff
@@ -1623,6 +1623,8 @@ func (db *Database) documentUpdateFunc(docExists bool, doc *Document, allowImpor
 	syncFnBody[BodyRev] = newRevID
 	if newDoc.Deleted {
 		syncFnBody[BodyDeleted] = true
+	} else {
+		doc.Deleted = false
```

Comment:
If we need to set doc.Deleted = false here, based on newDoc.Deleted, why don't we need to set doc.Deleted = true when newDoc.Deleted = true?  More to the point, the underlying flags have already been set (in updateWinningRevAndSetDocFlags).  I would have expected that we would no longer be using doc.Deleted by this point, as it's one of those transport-only properties, and doesn't reflect the actual syncData state.

Reply:
Probably better to step through in Sync up if possible but this was required in the import case. We get the old doc and update it, however, it appears that in the import case we're not re-setting this and so if the old rev was deleted the new rev won't be 'un-deleted'.